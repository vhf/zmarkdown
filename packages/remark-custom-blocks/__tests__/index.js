import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'

const render = (text, allowTitle) => unified()
  .use(reParse)
  .use(remark2rehype)
  .use(plugin, {
    secret: {
      classes: 'spoiler',
    },
    s: {
      classes: 'spoiler',
    },
    information: {
      classes: 'information ico-after',
    },
    i: {
      classes: 'information ico-after',
    },
    question: {
      classes: 'question ico-after',
    },
    q: {
      classes: 'question ico-after',
    },
    attention: {
      classes: 'warning ico-after',
    },
    a: {
      classes: 'warning ico-after',
    },
    erreur: {
      classes: 'error ico-after',
    },
    e: {
      classes: 'error ico-after',
    },
    neutre: {
      classes: 'neutral foo',
      title: 'required',
    },
    customizableBlock: {
      classes: 'neutral foo',
      title: 'optional',
    },
  }, allowTitle)
  .use(stringify)
  .processSync(text)

const fixture = dedent`
  [[s]]
  | Secret Block

  [[s]]
  |Secret Block

  [[secret]]
  | another

  > [[s]]
  > | > Blockquote in secret block in blockquote

  [[i]]
  | Information Block

  [[information]]
  | another

  [[q]]
  | Question Block

  [[question]]
  | another

  [[a]]
  | Attention Block

  [[attention]]
  | another

  [[e]]
  | Erreur Block

  [[erreur]]
  | another

  [[se]]
  | not a block

  [[secretsecret]]
  | not a block

  [[SECRET]]
  | not a block

  [[s]]
  | Multiline block
  |
  | > with blockquote !

  | Not a block

  [[attention | title]]
  | not parsed
`

test('Common', () => {
  const {contents} = render(fixture)
  expect(contents).toMatchSnapshot()
})

test('title is required', () => {
  const {contents} = render(dedent`
    [[neutre]]
    | no

    [[neutre|my title]]
    | yes

    [[neutre |  my **title**]]
    | yes
  `)
  expect(contents).toMatchSnapshot()
})

test('title is optional', () => {
  const {contents} = render(dedent`
    [[customizableBlock]]
    | yes

    [[customizableBlock    | my <i>t√Øtle</i>]]
    | yes
  `)
  expect(contents).toMatchSnapshot()
})

test('regression 1', () => {
  const {contents} = render(dedent`
    content before
    [[s]]
    |Block
    with content after
  `)
  expect(contents).toMatchSnapshot()
})

test('Errors without config', () => {
  const fail = () => unified()
    .use(reParse)
    .use(remark2rehype)
    .use(plugin)
    .use(stringify)
    .processSync('')

  expect(fail).toThrowError(Error)
})
