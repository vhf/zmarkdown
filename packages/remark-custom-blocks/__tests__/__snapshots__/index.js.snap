// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common 1`] = `
"<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><p>Secret Block</p></div></div>
<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><p>Secret Block</p></div></div>
<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><p>another</p></div></div>
<blockquote>
<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><blockquote>
<p>Blockquote in secret block in blockquote</p>
</blockquote></div></div>
</blockquote>
<div class=\\"custom-block information ico-after\\"><div class=\\"custom-block-body\\"><p>Information Block</p></div></div>
<div class=\\"custom-block information ico-after\\"><div class=\\"custom-block-body\\"><p>another</p></div></div>
<div class=\\"custom-block question ico-after\\"><div class=\\"custom-block-body\\"><p>Question Block</p></div></div>
<div class=\\"custom-block question ico-after\\"><div class=\\"custom-block-body\\"><p>another</p></div></div>
<div class=\\"custom-block warning ico-after\\"><div class=\\"custom-block-body\\"><p>Attention Block</p></div></div>
<div class=\\"custom-block warning ico-after\\"><div class=\\"custom-block-body\\"><p>another</p></div></div>
<div class=\\"custom-block error ico-after\\"><div class=\\"custom-block-body\\"><p>Erreur Block</p></div></div>
<div class=\\"custom-block error ico-after\\"><div class=\\"custom-block-body\\"><p>another</p></div></div>
<p>[[se]]
| not a block</p>
<p>[[secretsecret]]
| not a block</p>
<p>[[SECRET]]
| not a block</p>
<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><p>Multiline block</p><blockquote>
<p>with blockquote !</p>
</blockquote></div></div>
<p>| Not a block</p>
<p>[[attention | title]]
| not parsed</p>"
`;

exports[`regression 1 1`] = `
"<p>content before</p>
<div class=\\"custom-block spoiler\\"><div class=\\"custom-block-body\\"><p>Block</p></div></div>
<p>with content after</p>"
`;

exports[`title is optional 1`] = `
"<div class=\\"custom-block neutral foo\\"><div class=\\"custom-block-body\\"><p>yes</p></div></div>
<div class=\\"custom-block neutral foo\\"><div class=\\"custom-block-heading\\">my t√Øtle</div><div class=\\"custom-block-body\\"><p>yes</p></div></div>"
`;

exports[`title is required 1`] = `
"<p>[[neutre]]
| no</p>
<div class=\\"custom-block neutral foo\\"><div class=\\"custom-block-heading\\">my title</div><div class=\\"custom-block-body\\"><p>yes</p></div></div>
<div class=\\"custom-block neutral foo\\"><div class=\\"custom-block-heading\\">my <strong>title</strong></div><div class=\\"custom-block-body\\"><p>yes</p></div></div>"
`;
